scene/gui/range.cpp

```diff
@@ -95,6 +95,9 @@ void Range::set_value(double p_val) {
 
 void Range::_set_value_no_signal(double p_val) {
 	if (!Math::is_finite(p_val)) {
+		if (allow_non_finite) {
+			shared->val = p_val;
+		}
 		return;
 	}
```

Comment:
Just because non-finite is allowed doesn't mean you can set arbitrary values without checks. For example, `+INF` should be limited to `max_value` if `allow_greater` is false.

Reply:
Is this ok? 
```
if (!Math::is_finite(p_val) && !allow_non_finite) {
	return;
}
```

Reply:
I think yes, since there are currently no negative checks in the code below. All comparisons with NaN return false (except `!=`), rounding NaN returns NaN. But probably the following code would be more reliable in case we accidentally add a negative check in the future:

```c++
if (!Math::is_finite(p_val)) {
    if (allow_non_finite) {
        if (Math::is_nan(p_val)) {
            shared->val = p_val;
            return;
        }
    } else {
        return;
    }
}
```